ghcide/src/Development/IDE/Spans/Common.hs

```diff
@@ -29,13 +30,15 @@ import           GHC.Generics
 import           System.FilePath
 
 import           Control.Lens
+import           Data.IntMap                  (IntMap)
 import           Development.IDE.GHC.Compat
 import           Development.IDE.GHC.Orphans  ()
 import qualified Language.LSP.Protocol.Lens   as JL
 import           Language.LSP.Protocol.Types
 
 type DocMap = NameEnv SpanDoc
 type TyThingMap = NameEnv TyThing
+type ArgDocMap = NameEnv (IntMap SpanDoc)
```

Comment:
Instead of creating a new `ArgDocMap`, an alternative implementation is to extend `DocMap` to `NameEnv (SpanDoc, IntMap SpanDoc)`.

Reply:
That sounds sensible to me? Any reason not to do that?

Reply:
Looking at the code it seems like it would simplifiy things quite a lot.

Reply:
Now I have second thoughts about this because the alternative implementation `type DocMap = NameEnv (SpanDoc, IntMap SpanDoc)` is semantically incorrect.  The semantically correct version should be `type DocMap = NameEnv (Maybe SpanDoc, IntMap SpanDoc)` or `type DocMap = NameEnv (Maybe SpanDoc, Maybe (IntMap SpanDoc))` because a `Name` can have only function doc or arg doc.

I prefer to keep the old implementation, i.e., introducing a new `ArgDoc`, since the semantically correct version of the alternative implementation seems more complicated.